generator_process/actions/prompt_to_image.py

```diff
@@ -236,6 +236,25 @@ def approximate_decoded_latents(latents):
 
     return latents_ubyte.numpy()
 
+def model_snapshot_folder(model, preferred_revision: str | None = None):
+    """ Try to find the preferred revision, but fallback to another revision if necessary. """
+    storage_folder = model
+    if os.path.exists(os.path.join(storage_folder, 'model_index.json')): # converted model
+        snapshot_folder = storage_folder
+    else: # hub model
+        ref_path = None
+        for revision in os.listdir(os.path.join(storage_folder, "refs")):
```

Comment:
When `preferred_revision` is None it's expecting a fp32 model, but this will lead to using whichever is the last revision depending on how the os returns files from listdir(). This may lead to incorrect behavior when multiple revisions are downloaded.

Reply:
I think loading the `fp16` revision should still work, even if half precision is disabled.